package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -629,10 +659,35 @@ def _read_next_timestep(self):
             fields = f.readline().split()
             if ids:
                 indices[i] = fields[attr_to_col_ix["id"]]
-            ts.positions[i] = [fields[dim] for dim in coord_cols]
+            coords = [fields[dim] for dim in coord_cols]
+            if len(coords) > 3:
+                if self._unwrap:
+                    if not triclinic:
+                        coords = np.array(coords,np.float)
+                        coords = [coords[x]+ts.dimensions[x]*coords[x+3] for x in range(3)]
```

Comment:
This looks like it could scale badly.  If we convert `coords` to a numpy array, we can check its shape and work wholly in numpy... something like:
```python
coords = np.array(coords)

if coords.shape[1] == 6:  # also check it's either 3 or 6, else we've got something weird
    images = coords[:, 3:]
    coords = coords[:, :3]
    coords += images * ts.dimensions[:3]
```